viewing_party/party.py

```diff
@@ -0,0 +1,196 @@
+def create_movie(movie_title, genre, rating):
+
+    if movie_title and genre and rating:
+        return {"title" : movie_title, "genre" : genre, "rating" : rating}
+    return None
+   
+
+def add_to_watched(user_data, movie):
+  
+    user_data["watched"].append(movie)
+    return user_data
+   
+def add_to_watchlist(user_data, movie):
+    user_data["watchlist"].append(movie)
+    return user_data
+
+def watch_movie(janes_data, Title_A):
+
+    for each in janes_data["watchlist"]:
+
+         if Title_A in each.values():
+            janes_data["watched"].append(janes_data["watchlist"].pop())
+         
+
+    return janes_data
+    
+
+def get_watched_avg_rating(janes_data):
+    
+    sum_counter = 0.0
+    for each in janes_data["watched"]:
+           sum_counter += each["rating"]
+    if len(janes_data["watched"]):
+
+         return sum_counter/len(janes_data["watched"])
+    else:
+        return 0
+
+
+def get_most_watched_genre(janes_data):
+  genre_dict = {}
+    
+  for listi in janes_data["watched"]:
+        for key, val in listi.items():           
+            if key=="genre":               
+                if val not in genre_dict:
+                  genre_dict[val]=1         
+                else:
+                  genre_dict[val] +=1
+    
+  counter = 0
+  genree = None
+  for key, val in genre_dict.items():
+    if val > counter:
+      counter = val
+      genree = key 
+  return genree
+    
+def get_unique_watched(amandas_data):
+
+    set_watched = set()
+    set_friends = set()
+
+    for each in amandas_data["watched"]:
+        set_watched.add(each["title"])
+  
+    
+    for each_friends in amandas_data["friends"]:
+        inner_list_values = each_friends["watched"]  # this is dictionaries collected as a list as values of a key ("watched")
+        for j in range(len(inner_list_values)):   
+            set_friends.add(inner_list_values[j]["title"])
+
+    unique_set  = set_watched - set_friends
+    new_unique_list = list(unique_set)
+    print(unique_set)
+    return_list = []
+    
+    
+    for j in range(len(new_unique_list)):
+        unique_dictionaries = {}
+        unique_dictionaries["title"] = new_unique_list[j]
+        return_list.append(unique_dictionaries)
+
+    return return_list
+
+def get_friends_unique_watched(amandas_data):
+    set_watched = set()
+    set_friends = set()
+
+    for each in amandas_data["watched"]:
+        set_watched.add(each["title"])
+ 
+    
+    for each_friends in amandas_data["friends"]:
+        inner_list_values = each_friends["watched"]  # this is dictionaries collected as a list as values of a key ("watched")
+        for j in range(len(inner_list_values)):   
+            set_friends.add(inner_list_values[j]["title"])
```

Comment:
This code is repeated on lines 125-128 and 157-160. A helper function such as `create_set_friends` to encapsulate this functionality could help DRY (don't repeat yourself) up your code.